cli/cfncluster/cfnconfig.py

```diff
@@ -268,47 +270,133 @@ def __init__(self, args):
-                        if self.__sanity_check and self.__ebs_options.get(key)[1] is not None:
-                            config_sanity.check_resource(self.region,self.aws_access_key_id, self.aws_secret_access_key,
-                                                self.__ebs_options.get(key)[1],__temp__)
-                        self.parameters.append((self.__ebs_options.get(key)[0],__temp__))
+                        __temp_dir_list.append(__temp__)
+
                     except configparser.NoOptionError:
                         pass
+                    except configparser.NoSectionError:
+                        print("ERROR: [%s] section defined in [ebs_settings] does not exist" %section)
+                        sys.exit(1)
         except AttributeError:
             pass
 
+        try:
+            if len(__temp_dir_list) == len(self.__ebs_section):
+                self.parameters.append(('SharedDir', ",".join(__temp_dir_list)))
+            # for backwards compatibility with just 1 volume explicitly specified through ebs_settings
+            elif len(self.__ebs_section) == 1:
+                try:
+                    __temp_shared_dir = __config.get(self.__cluster_section, 'shared_dir')
+                    if not __temp_shared_dir:
+                        print("ERROR: shared_dir defined but not set")
+                        sys.exit(1)
+                    self.parameters.append(('SharedDir', str(__temp_shared_dir)))
+                except configparser.NoOptionError:
+                    pass
+            else:
+                print("ERROR: not enough mount points provided.\n"
+                      "When using multiple EBS Volumes, please specify a [mount_point] under each [EBS setting] section")
+                sys.exit(1)
+        except AttributeError:
+            try:
+                __temp_shared_dir = __config.get(self.__cluster_section, 'shared_dir')
+                if not __temp_shared_dir:
+                    print("ERROR: shared_dir defined but not set")
+                    sys.exit(1)
+                self.parameters.append(('SharedDir', str(__temp_shared_dir)))
+            except configparser.NoOptionError:
+                pass
+
+        # Dictionary list of all EBS options
+        self.__ebs_options = dict(ebs_snapshot_id=('EBSSnapshotId','EC2Snapshot'), volume_type=('VolumeType', None),
+                                  volume_size=('VolumeSize',None), ebs_kms_key_id=('EBSKMSKeyId', None),
+                                  volume_iops=('VolumeIOPS',None), encrypted=('EBSEncryption', None),
+                                  ebs_volume_id=('EBSVolumeId','EC2Volume'))
+        # EBS options processing
+        try:
+            if self.__ebs_section:
+                for key in self.__ebs_options:
+                    __temp_parameter_list = []
+                    for section in self.__ebs_section:
+                        try:
+                            __temp__ = __config.get(section, key)
```

Comment:
why the double trailing underscore? could we have a more meaningful name rather then temp only, especially cause this var is used in several places?